test/service/test_request_validation.py

```diff
@@ -236,45 +268,203 @@ def test_file_order(self):
+            ('foo bar', 'Invalid characters within parameter')
         ]:
             response = requests.get(url=url, params={'catalog': catalog})
             self.assertEqual(400, response.status_code, response.json())
-            self.assertIn(error, response.json()['Message'])
+            self.assertEqual(1, len(response.json()['invalid_parameters']))
+            self.assertIn(error, response.json()['invalid_parameters'][0]['message'])
+
+    def test_deterministic_response(self):
+        url = furl(url=self.base_url, path='/index/files').url
+        shuffle_parameters = [
+            ('filters', '{"}'),
+            ('catalog', 'foo'),
+            ('sort', 'bar'),
+            ('order', 'asc')]
+        shuffle(shuffle_parameters)
+        params = {parameter[0]: parameter[1] for parameter in shuffle_parameters}
+        response = requests.get(url=url, params=params)
+        self.assertEqual(400, response.status_code, response.json())
+        self.assertEqual(3, len(response.json()['invalid_parameters']))
+        self.assertEqual('catalog', response.json()['invalid_parameters'][0]['name'])
+        self.assertEqual('filters', response.json()['invalid_parameters'][1]['name'])
+        self.assertEqual('sort', response.json()['invalid_parameters'][2]['name'])
+
+    @unittest.skip('https://github.com/DataBiosphere/azul/issues/2465')
+    def test_missing_uuid(self):
+        url = furl(url=self.base_url, path='/fetch/repository/files/').url
+        response = requests.get(url, params={'replica': 'aws'})
+        self.assertEqual(400, response.status_code)
+
+    def test_default_for_missing_params(self):
+        path = '/test/mock/endpoints'
+        method_spec = {'parameters': [],
+                       'responses': {'200': {'description': 'OK'}}, 'tags': ['Index']}
+
+        for required, catalog in product((True, False), (self.catalog, None)):
+            param_spec = {'in': 'query', 'schema': {
+                'type': 'string',
+                'pattern': '^([a-z0-9]{1,64})$',
+                'enum': ['test']
+            }, 'required': required, 'name': f'required-{required}-default-{catalog}'}
+            if catalog is not None:
+                param_spec['schema']['default'] = catalog
+            method_spec['parameters'].append(param_spec)
+
+        @self.app_module.app.route(path,
+                                   validate=True,
+                                   path_spec=None,
+                                   method_spec=method_spec,
+                                   methods=['GET'])
+        def test_method():
+            return dict(self.app_module.app.current_request.query_params)
+
+        url = furl(url=self.base_url, path=path).url
+        response = requests.get(url=url)
+        self.assertEqual(400, response.status_code, response.json())
+        self.assertEqual('required-True-default-None',
+                         response.json()['missing_parameters'][0]['name'])
```

Comment:
Using one (from itertools) would remove the need for the assert after this one.

```suggestion
        self.assertEqual('required-True-default-None',
                         one(response.json()['missing_parameters'])['name'])
```